include/fmt/core.h

```diff
@@ -877,16 +881,18 @@ FMT_CONSTEXPR typename internal::result_of<Visitor(int)>::type
 template <typename Char, typename ErrorHandler = internal::error_handler>
 class basic_parse_context : private ErrorHandler {
  private:
-  basic_string_view<Char> format_str_;
-  int next_arg_id_;
+   basic_string_view<Char> primary_format_str_;
```

Comment:
Why is `primary_format_str_` needed? Shouldn't `format_str_` be enough to resolve references?

Also the indent is a bit off.

Reply:
AFAIU, while parsing/formatting, `basic_parse_context::advance_to()` can be called, which calls `format_str_.remove_prefix()`, which 'invalidates' `format_str_` and `arg_ref::name` (which is `string_view_metadata`) can not be created because it needs primary format string to calculate offset from the beginning of the format. Does that make sense, or I just messed up something?

Reply:
Right now it is not a problem because we agreed to address custom formatting separately.

Reply:
So `basic_parse_context::advance_to()` is called only for custom formatters?

Reply:
More or less. The whole context machinery is there to support extension mechanism (custom formatters).